src/app/projects/add-edit-project/add-edit-project.component.ts

```diff
@@ -549,19 +552,21 @@ export class AddEditProjectComponent implements OnInit, AfterViewInit, OnDestroy
    * @returns {FormGroup[]} The array of shapefile form groups.
    */
     buildShapefilesFormArray(projectData: Project): FormGroup[] {
+      let shapefilesToFillFormWith = this.shapefileDocuments as unknown as ProjectShapefileOrDocument[];
+      
       if (Array.isArray(projectData.shapefiles)) {
-        return projectData.shapefiles.map(shapefile => {
-          return new FormGroup({
-            'document': new FormControl(shapefile.document),
-            'documentFileName': new FormControl(shapefile.documentFileName),
-            'title': new FormControl(shapefile.title),
-            'colour': new FormControl(shapefile.colour),
-            'order': new FormControl(shapefile.order)
-          })
-        })
-      } else {
-        return [];
+        shapefilesToFillFormWith = projectData.shapefiles;
       }
+
+      return shapefilesToFillFormWith.map(shapefile => {
+        return new FormGroup({
+          'document': new FormControl(shapefile?.document || shapefile?.id),
```

Comment:
Is this where you're combining shape files from different sources (including legacy)?

Reply:
Yeah so this function builds forms out of shapefile data, which could be coming from the route resolver or from `project.shapefiles`. I modified the function a bit so it could handle both source of shapefiles, which differ in their types slightly